pages/editpage/[id].js

```diff
@@ -1,5 +1,55 @@
-import CreateRecipe from "../../components/Form";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import EditRecipe from "../../components/Form/Edit";
 
-export default function CreatePage() {
-  return <CreateRecipe />;
+async function fetchRecipe(id) {
+  try {
+    const response = await fetch(`/api/recipes/${id}`);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error fetching recipe:", error);
+    return null;
+  }
+}
+
+export default function EditRecipePage() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [initialRecipe, setInitialRecipe] = useState(null);
+
+  useEffect(() => {
+    async function fetchInitialRecipe() {
+      const recipe = await fetchRecipe(id);
+      setInitialRecipe(recipe);
+    }
+
+    fetchInitialRecipe();
+  }, [id]);
```

Comment:
Seeing as you have `swr` installed in this project, i think it makes sense to use that to fetch the data. 

```suggestion
 const { data: initialRecipe, isLoading } = useSWR(`/api/recipes/${id}`);
```

We would need to rename the destructured `data` to be `initialRecipe` and we can also get access to a `isLoading` state from `swr`.